test/lib/stack.spec.js

```diff
@@ -22,66 +20,76 @@ describe("stack proxy", () => {
   })
 
-  describe("when initialized with an cozyURL + token", () => {
-    
+  describe('when initialized with an cozyURL + token', () => {
     const params = { cozyURL, token, onCreateApp, onDeleteApp }
-    
+
     beforeAll(() => {
       jest.clearAllMocks()
-      
+
       stack.init(params)
       stack.get.app()
     })
 
-    it("should call the internal stack init", () => {
-      expect( internal.init ).toHaveBeenCalled()
+    it('should call the internal stack init', () => {
+      expect(internal.init).toHaveBeenCalled()
     })
 
-    it("should not call the cozy-client stack init", () => {
-      expect( client.init ).not.toHaveBeenCalled()
+    it('should not call the cozy-client stack init', () => {
+      expect(client.init).not.toHaveBeenCalled()
     })
 
-    it("should forward requests to the internal stack client", () => {
-      expect( internal.get.app ).toHaveBeenCalled()
+    it('should forward requests to the internal stack client', () => {
+      expect(internal.get.app).toHaveBeenCalled()
     })
 
-    it("should not forward requests to the cozy-client stack client", () => {
-      expect( client.get.app ).not.toHaveBeenCalled()
+    it('should not forward requests to the cozy-client stack client', () => {
+      expect(client.get.app).not.toHaveBeenCalled()
     })
-
   })
 
-  describe("when initialized with a cozy-client instance", () => {
-
-    const cozyClient = { 
+  describe('when initialized with a cozy-client instance', () => {
+    const cozyClient = {
       getStackClient: () => {
         return {
-          token: { token: "mytoken"},
-          uri: "https://test.mycozy.cloud"
+          token: { token: 'mytoken' },
+          uri: 'https://test.mycozy.cloud'
         }
       }
     }
     const params = {
       cozyClient,
       onCreateApp: function() {},
-      onDeleteApp: function() {},
+      onDeleteApp: function() {}
     }
 
     beforeAll(() => {
       jest.clearAllMocks()
-      
+
       stack.init(params)
       stack.get.app()
     })
 
-    it("should call the client stack init", () => {
-      expect( client.init ).toHaveBeenCalled()
+    it('should call the client stack init', () => {
+      expect(client.init).toHaveBeenCalled()
     })
 
-    it("should forward requests to the cozy-client stack client", () => {
-      expect( client.get.app ).toHaveBeenCalled()
+    it('should forward requests to the cozy-client stack client', () => {
+      expect(client.get.app).toHaveBeenCalled()
     })
-    
   })
 
+  it('should throw error if trying to use the stack before init', () => {
+    jest.clearAllMocks()
+    const params = { cozyURL, token, onCreateApp, onDeleteApp }
+
+    // TODO for now the stack is default to internal even before init
+    // expect(() => {
+    //   stack.getStack()
+    // }).toThrowErrorMatchingSnapshot()
+    stack.init(params)
+    expect(() => {
+      stack.getStack()
+    }).not.toThrowError()
+    expect(stack.getStack()).toMatchSnapshot()
```

Comment:
Pourquoi compares-tu un snapshot ? Ca ne correspond pas au test décrit. Que cherches-tu à vérifier / confirmer ?

Reply:
Je voulais vérifier le fait que la stack est correctement défini et j'ai jugé plus malin de placé un snapshot qui permet aussi de vérifier qu'il a les bonnes propriétés au passage, mais je me trompe peut être un est `.toBeDefined()` pourrait suffire.

Reply:
Le snapshot c'est un peu l'arme atomique. Ca fait des tests qui vont casser à la moindre amélioration. Si ton test c'est "vérifier que le prototype n'a pas changé" alors vas-y pour un snapshot.

Si le sujet c'est 'should throw error if trying to use the stack before init' alors tu vérifies que ça ne jette pas d'erreur et c'est tout.

Là tu as un test qui dit "je vérifie que ça ne jette pas d'erreur si j'utilise la stack avant l'init" mais : 
- Tu vérifies l'inexistance d'erreur *après* l'init
- Tu vérifies que le résultat de l'init reste constant

Bref, ça ne correspond pas.

Reply:
D'accord, je supprime cette partie pour que ça corresponde au test du coup